spec/lib/services/redfish-validator-service-spec.js

```diff
@@ -63,5 +69,79 @@ describe("Redfish Validator Service", function() {
             });
     });
 
+    describe("getVendorNameById", function() {
 
+        beforeEach('getVendorNameById beforeEach', function(){
+            waterline.nodes.getNodeById.reset();
+        });
+
+        it('should get Cisco vendor name by identifier', function() {
+            var id = "599337d6ff99ed24305bc58a";
+            var sampleDataInfo = {
+                "id": id,
+                "identifiers": [
+                    "1.1.1.0:sys/rack-unit-2"
+                ]
+            };
+
+            waterline.nodes.getNodeById.resolves(sampleDataInfo);
+            return redfish.getVendorNameById(id)
+                .then(function(result){
+                    expect(result.vendor).to.equal("Cisco");
+                    expect(result.node).to.deep.equal(sampleDataInfo);
+                    expect(waterline.nodes.getNodeById).to.be.calledOnce;
+                    expect(waterline.nodes.getNodeById).to.be
```

Comment:
I would suggest you don't make assertion on the `waterline.nodes.getNodeById` as the function `getVendorNameById` is designed to get a vendor name by parsing the node id, but it doesn't need to depend on the waterline to do such parsing. For example, in future, we may can get the vendor name directly from the SKU, after then, we don't need to change unit-test